localstack/services/apigateway/helpers.py

```diff
@@ -1172,26 +1172,25 @@ def add_path_methods(rel_path: str, parts: List[str], parent_id=""):
 
             # Create the `IntegrationResponse` for the previously created `Integration`
             if method_integration_responses := method_integration.get("responses"):
-                default_method_integration_response = method_integration_responses.get(
-                    "default", {}
-                )
-                integration_response_templates = default_method_integration_response.get(
-                    "responseTemplates"
-                )
-                integration_response_parameters = default_method_integration_response.get(
-                    "responseParameters"
-                )
+                for pattern, integration_responses in method_integration_responses.items():
+                    method_integration_response = method_integration_responses.get(pattern, {})
```

Comment:
nit: no need to get the value from the `integration_responses` dict here, as we already extracted it via the iterator in the line above (`integration_responses` should be equivalent to `method_integration_response`)

Reply:
Nice catch 😺 !!